pycbc/inference/sampler/multinest.py

```diff
@@ -321,6 +329,61 @@ def checkpoint(self):
         if not checkpoint_valid:
             raise IOError("error writing to checkpoint file")
 
+    def setup_output(self, output_file, force=False, injection_file=None):
+        """Sets up the sampler's checkpoint and output files.
+
+        The checkpoint file has the same name as the output file, but with
+        ``.checkpoint`` appended to the name. A backup file will also be
+        created.
+
+        If the output file already exists, an ``OSError`` will be raised.
+        This can be overridden by setting ``force`` to ``True``.
+
+        Parameters
+        ----------
+        sampler : sampler instance
+            Sampler
+        output_file : str
+            Name of the output file.
+        force : bool, optional
+            If the output file already exists, overwrite it.
+        injection_file : str, optional
+            If an injection was added to the data, write its information.
+        """
+        # check for backup file(s)
+        checkpoint_file = output_file + '.checkpoint'
+        backup_file = output_file + '.bkup'
+        if not self.is_main_process:
+            self.checkpoint_file = checkpoint_file
+            self.backup_file = backup_file
+            self.checkpoint_valid = True
+            self.new_checkpoint = True
+        else:
+            # check if we have a good checkpoint and/or backup file
+            logging.info("Looking for checkpoint file")
+            checkpoint_valid = validate_checkpoint_files(checkpoint_file,
+                                                         backup_file)
+            # Create a new file if the checkpoint doesn't exist, or if it is
+            # corrupted
+            self.new_checkpoint = False  # tracks whether this is a new file
+            if not checkpoint_valid:
+                logging.info("Checkpoint not found or not valid")
+                create_new_output_file(self, checkpoint_file, force=force,
+                                       injection_file=injection_file)
+                # now the checkpoint is valid
+                self.new_checkpoint = True
+                # copy to backup
+                shutil.copy(checkpoint_file, backup_file)
+            # write the command line, startup
+            for f_n in [checkpoint_file, backup_file]:
+                with self.io(f_n, "a") as f_p:
+                    f_p.write_command_line()
+                    f_p.write_resume_point()
+            # store
+            self.checkpoint_file = checkpoint_file
+            self.backup_file = backup_file
+            self.checkpoint_valid = checkpoint_valid
+
```

Comment:
I'd rather not duplicate this much code. Make `setup_output` an abstract method in `BaseSampler` instead, and move it's current functionality to a stand-alone function (you can stick it in the "Convenience functions" section in `sampler/base.py`, just before `create_new_output_file`). Then add a `setup_output` to `BaseMCMC` (which should take care of both `emcee` and `emcee-pt`) and `CPNest` that just calls the stand alone function. Here, run your check for `is_main_process`, calling the stand-alone function otherwise.